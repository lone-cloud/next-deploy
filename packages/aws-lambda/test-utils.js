const fse = require('fs-extra');
const path = require('path');
const os = require('os');
const LambdaComponent = require('./serverless');

const createTmpDir = () => {
  return fse.mkdtemp(path.join(os.tmpdir(), 'test-'));
};

const createComponent = async () => {
  // create tmp folder to avoid state collisions between tests
  const tmpFolder = await createTmpDir();

  const component = new LambdaComponent('TestLambda', {
    stateRoot: tmpFolder,
  });

  await component.init();

  return component;
};

module.exports = { createComponent, createTmpDir };
