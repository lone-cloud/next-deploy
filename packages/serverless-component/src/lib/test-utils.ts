import fse from 'fs-extra';
import path from 'path';

import { BUILD_DIR } from '../constants';

export const cleanupFixtureDirectory = (fixtureDir: string) => (): Promise<void> => {
  return fse.remove(path.join(fixtureDir, BUILD_DIR));
};
