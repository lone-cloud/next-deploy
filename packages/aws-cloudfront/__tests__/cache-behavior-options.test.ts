import { createComponent } from './test-utils';

import { mockCreateDistribution, mockCreateDistributionPromise } from 'aws-sdk';
import CloudFrontComponent from '../serverless';

jest.mock('aws-sdk', () => require('../__mocks__/aws-sdk.mock'));

describe('Input origin as a custom url', () => {
  let component: CloudFrontComponent;

  beforeEach(async () => {
    mockCreateDistributionPromise.mockResolvedValueOnce({
      Distribution: {
        Id: 'distribution123',
      },
    });

    component = await createComponent();
  });

  it('creates distribution with custom default behavior options', async () => {
    await component.default({
      defaults: {
        ttl: 0,
        forward: {
          headers: ['Accept', 'Accept-Language'],
          cookies: 'all',
          queryString: true,
          queryStringCacheKeys: ['query'],
        },
        allowedHttpMethods: ['GET', 'HEAD', 'OPTIONS', 'PUT', 'POST', 'PATCH', 'DELETE'],
        viewerProtocolPolicy: 'https-only',
        smoothStreaming: true,
        compress: true,
        fieldLevelEncryptionId: '123',
      },
      origins: ['https://mycustomorigin.com'],
    });

    expect(mockCreateDistribution.mock.calls[0][0]).toMatchSnapshot();
  });

  it('creates distribution with custom behavior options', async () => {
    await component.default({
      defaults: {
        ttl: 0,
      },
      origins: [
        {
          url: 'https://mycustomorigin.com',
          pathPatterns: {
            '/sample/path': {
              ttl: 0,
              forward: {
                headers: 'all',
                cookies: ['auth-token'],
                queryString: true,
              },
              allowedHttpMethods: ['GET', 'HEAD'],
              viewerProtocolPolicy: 'redirect-to-https',
              compress: false,
              fieldLevelEncryptionId: '321',
            },
          },
        },
      ],
    });

    expect(mockCreateDistribution.mock.calls[0][0]).toMatchSnapshot();
  });
});
