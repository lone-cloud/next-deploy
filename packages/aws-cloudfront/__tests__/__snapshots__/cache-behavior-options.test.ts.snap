// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input origin as a custom url creates distribution with custom behavior options 1`] = `
Object {
  "DistributionConfig": Object {
    "Aliases": Object {
      "Items": Array [],
      "Quantity": 0,
    },
    "CacheBehaviors": Object {
      "Items": Array [
        Object {
          "AllowedMethods": Object {
            "CachedMethods": Object {
              "Items": Array [
                "GET",
                "HEAD",
              ],
              "Quantity": 2,
            },
            "Items": Array [
              "GET",
              "HEAD",
            ],
            "Quantity": 2,
          },
          "Compress": false,
          "DefaultTTL": 0,
          "FieldLevelEncryptionId": "321",
          "ForwardedValues": Object {
            "Cookies": Object {
              "Forward": "whitelist",
              "WhitelistedNames": Object {
                "Items": Array [
                  "auth-token",
                ],
                "Quantity": 1,
              },
            },
            "Headers": Object {
              "Items": Array [
                "*",
              ],
              "Quantity": 1,
            },
            "QueryString": true,
            "QueryStringCacheKeys": Object {
              "Items": Array [],
              "Quantity": 0,
            },
          },
          "LambdaFunctionAssociations": Object {
            "Items": Array [],
            "Quantity": 0,
          },
          "MaxTTL": 0,
          "MinTTL": 0,
          "PathPattern": "/sample/path",
          "SmoothStreaming": false,
          "TargetOriginId": "mycustomorigin.com",
          "TrustedSigners": Object {
            "Enabled": false,
            "Quantity": 0,
          },
          "ViewerProtocolPolicy": "redirect-to-https",
        },
      ],
      "Quantity": 1,
    },
    "CallerReference": "1566599541192",
    "Comment": "",
    "DefaultCacheBehavior": Object {
      "AllowedMethods": Object {
        "CachedMethods": Object {
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Items": Array [
          "HEAD",
          "GET",
        ],
        "Quantity": 2,
      },
      "Compress": false,
      "DefaultTTL": 0,
      "FieldLevelEncryptionId": "",
      "ForwardedValues": Object {
        "Cookies": Object {
          "Forward": "none",
        },
        "Headers": Object {
          "Items": Array [],
          "Quantity": 0,
        },
        "QueryString": false,
        "QueryStringCacheKeys": Object {
          "Items": Array [],
          "Quantity": 0,
        },
      },
      "LambdaFunctionAssociations": Object {
        "Items": Array [],
        "Quantity": 0,
      },
      "MaxTTL": 31536000,
      "MinTTL": 0,
      "SmoothStreaming": false,
      "TargetOriginId": "mycustomorigin.com",
      "TrustedSigners": Object {
        "Enabled": false,
        "Items": Array [],
        "Quantity": 0,
      },
      "ViewerProtocolPolicy": "redirect-to-https",
    },
    "Enabled": true,
    "HttpVersion": "http2",
    "Origins": Object {
      "Items": Array [
        Object {
          "CustomHeaders": Object {
            "Items": Array [],
            "Quantity": 0,
          },
          "CustomOriginConfig": Object {
            "HTTPPort": 80,
            "HTTPSPort": 443,
            "OriginKeepaliveTimeout": 5,
            "OriginProtocolPolicy": "https-only",
            "OriginReadTimeout": 30,
            "OriginSslProtocols": Object {
              "Items": Array [
                "TLSv1.2",
              ],
              "Quantity": 1,
            },
          },
          "DomainName": "mycustomorigin.com",
          "Id": "mycustomorigin.com",
          "OriginPath": "",
        },
      ],
      "Quantity": 1,
    },
    "PriceClass": "PriceClass_All",
  },
}
`;

exports[`Input origin as a custom url creates distribution with custom default behavior options 1`] = `
Object {
  "DistributionConfig": Object {
    "Aliases": Object {
      "Items": Array [],
      "Quantity": 0,
    },
    "CacheBehaviors": Object {
      "Items": Array [],
      "Quantity": 0,
    },
    "CallerReference": "1566599541192",
    "Comment": "",
    "DefaultCacheBehavior": Object {
      "AllowedMethods": Object {
        "CachedMethods": Object {
          "Items": Array [
            "HEAD",
            "GET",
          ],
          "Quantity": 2,
        },
        "Items": Array [
          "GET",
          "HEAD",
          "OPTIONS",
          "PUT",
          "POST",
          "PATCH",
          "DELETE",
        ],
        "Quantity": 7,
      },
      "Compress": true,
      "DefaultTTL": 0,
      "FieldLevelEncryptionId": "123",
      "ForwardedValues": Object {
        "Cookies": Object {
          "Forward": "all",
        },
        "Headers": Object {
          "Items": Array [
            "Accept",
            "Accept-Language",
          ],
          "Quantity": 2,
        },
        "QueryString": true,
        "QueryStringCacheKeys": Object {
          "Items": Array [
            "query",
          ],
          "Quantity": 1,
        },
      },
      "LambdaFunctionAssociations": Object {
        "Items": Array [],
        "Quantity": 0,
      },
      "MaxTTL": 31536000,
      "MinTTL": 0,
      "SmoothStreaming": true,
      "TargetOriginId": "mycustomorigin.com",
      "TrustedSigners": Object {
        "Enabled": false,
        "Items": Array [],
        "Quantity": 0,
      },
      "ViewerProtocolPolicy": "https-only",
    },
    "Enabled": true,
    "HttpVersion": "http2",
    "Origins": Object {
      "Items": Array [
        Object {
          "CustomHeaders": Object {
            "Items": Array [],
            "Quantity": 0,
          },
          "CustomOriginConfig": Object {
            "HTTPPort": 80,
            "HTTPSPort": 443,
            "OriginKeepaliveTimeout": 5,
            "OriginProtocolPolicy": "https-only",
            "OriginReadTimeout": 30,
            "OriginSslProtocols": Object {
              "Items": Array [
                "TLSv1.2",
              ],
              "Quantity": 1,
            },
          },
          "DomainName": "mycustomorigin.com",
          "Id": "mycustomorigin.com",
          "OriginPath": "",
        },
      ],
      "Quantity": 1,
    },
    "PriceClass": "PriceClass_All",
  },
}
`;
